File: 2 - balanced random generator/python/balanced.py
import random


def unbalanced_generator():
	return random.choices([0, 1], [0.3, 0.7])[0]


def balanced_generator():
	while True:
		first_toss = unbalanced_generator()
		second_toss = unbalanced_generator()
		if first_toss != second_toss:
			return first_toss

File: 2 - balanced random generator/elixir/balanced.exs
defmodule Balanced do

	def unbalanced_generator() do
		Enum.random([0, 0, 0, 1, 1, 1, 1, 1, 1, 1])
	end

	def balanced_generator() do
		first_toss = unbalanced_generator()
		second_toss = unbalanced_generator()
		if first_toss != second_toss, do: first_toss, else: balanced_generator()
	end

end